rs/artifact_pool/src/canister_http_pool.rs

```diff
@@ -25,14 +27,16 @@ const POOL_CANISTER_HTTP_CONTENT: &str = "canister_http_content";
 
 type ValidatedCanisterHttpPoolSection = PoolSection<CanisterHttpResponseShare, ()>;
 
-type UnvalidatedCanisterHttpPoolSection = PoolSection<CanisterHttpResponseShare, ()>;
+type UnvalidatedCanisterHttpPoolSection =
+    PoolSection<CanisterHttpResponseShare, CanisterHttpResponseArtifact>;
```

Comment:
Just for my understanding: Would this also have worked?
```suggestion
    PoolSection<CanisterHttpResponseShare, Option<CanisterHttpResponse>>;
```

Reply:
Yeah that would have also worked. But I found the current version cleaner